### PR TITLE
add waterfall chart plugin reop

### PR DESCRIPTION
**Pull request by scsigma**
"waterfall-example" is the name of the new repo

### CODE REVIEW

**Comment by scsigma** [comment]
can someone review this plz

File: waterfall-example/README.md
Comment (Pearce-Ropion) [inline waterfall-example/README.md:1]: Lets clean up some of the boilerplate that was generated in this file

File: waterfall-example/src/App.test.js
Comment (Pearce-Ropion) [inline waterfall-example/src/App.test.js:1]: This looks like the auto-generated file. Either update the tests or remove it. If you remove this and don't add any additional tests, please remove all the testing packages from package.json too

File: waterfall-example/src/logo.svg
Comment (Pearce-Ropion) [inline waterfall-example/src/logo.svg:1]: Is this used?

File: waterfall-example/package-lock.json
Comment (Pearce-Ropion) [inline waterfall-example/package-lock.json:1]: You don't need both a `package-lock` and a `yarn.lock`. We use `yarn` company wide so lets delete this file (`package-lock.json`) and be sure to use `yarn` scripts when running this plugin.

File: waterfall-example/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "waterfall",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@sigmacomputing/plugin": "^1.0.5",
+    "@testing-library/jest-dom": "^5.16.5",
+    "@testing-library/react": "^13.4.0",
+    "@testing-library/user-event": "^13.5.0",
```
Comment (Pearce-Ropion) [inline waterfall-example/package.json:9]: If you aren't doing any testing, please remove these

File: waterfall-example/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    "highcharts-react-official": "^3.1.0",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "react-scripts": "5.0.1",
+    "web-vitals": "^2.1.4"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
```
Comment (Pearce-Ropion) [inline waterfall-example/package.json:20]: If you aren't doing any testing, remove the `test` command

File: waterfall-example/public/index.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (Pearce-Ropion) [inline waterfall-example/public/index.html:27]: Update this to `Sigma Waterfall Plugin`

File: waterfall-example/public/logo192.png
Comment (Pearce-Ropion) [inline waterfall-example/public/logo192.png:1]: Hmm, might want to ask design for some Sigma logo icons.

File: waterfall-example/public/logo512.png
Comment (Pearce-Ropion) [inline waterfall-example/public/logo512.png:1]: Hmm, might want to ask design for some Sigma logo icons.

File: waterfall-example/public/favicon.ico
Comment (Pearce-Ropion) [inline waterfall-example/public/favicon.ico:1]: Hmm, might want to ask design for some Sigma logo icons.

File: waterfall-example/src/App.js [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+const App = () => { 
+  const options = useMain();
+  return (
+    <div>
+      <HighchartsReact highcharts={Highcharts} options={options}/>
+    </div>
+  );
+}
+
+export default App;
```
Comment (Pearce-Ropion) [inline waterfall-example/src/App.js:255]: nit: newline

File: waterfall-example/src/App.css
Comment (Pearce-Ropion) [inline waterfall-example/src/App.css:1]: Are you using any of this?

File: waterfall-example/src/App.js [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+}
+
+
+const getSigmaData = (config, sigmaData) => {
+
+  // Async data conditional
+  if (!allSigmaDataReceived(config, sigmaData)) return null;
+
+  // This will tell us what to group by: day, week, month, year
+  let date_grouping = sigmaData[config["date grouping"]][0];
```
Comment (Pearce-Ropion) [inline waterfall-example/src/App.js:44]: `const`

File: waterfall-example/src/App.js [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+  // -------------------------------------------------------
+  // Sort the dates coming in to the data
+
+  // 1. combine the x and y arrays into an object
+  let arr_unsorted = []
+  for (let i = 0; i < sigmaData[config['x']].length; i++) {
+    arr_unsorted.push({'x': sigmaData[config['x']][i], 'y': sigmaData[config['y']][i]})
+  }
+
+  let sorted_arr = arr_unsorted.sort((a,b) => {
```
Comment (Pearce-Ropion) [inline waterfall-example/src/App.js:55]: `const`

File: waterfall-example/src/App.js [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+  sigmaData[config['y']] = sorted_arr.map((val) => val.y)
+
+  // Create the data object that fits into the highcharts series
+  // Call to attention how the formatting is for highcharts
+  const data = sigmaData[config['x']].map((date_input, i) => {
+    
+    let date = new Date(date_input)
+    let month_string = date.toDateString().split(' ')[1];
+    let day_num_string = date.toDateString().split(' ')[2];
+    let year_string = date.toDateString().split(' ')[3];
```
Comment (Pearce-Ropion) [inline waterfall-example/src/App.js:70]: `const`

Comment (Pearce-Ropion) [inline waterfall-example/src/App.js:70]:
```
const date = new Date(date_input);
const [, month, day, year] = date.toDateString().split(' ');
```

File: waterfall-example/src/App.js [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+        text: 'Profit'
+      }
+    },
+    tooltip: {
+      // Default Point format
+      // pointFormat: "<b>${point.y:,.2f}</b> USD"
+
+      // More Clean Tooltip formatter
+      enabled: true,
+      formatter: function () {
```
Comment (Pearce-Ropion) [inline waterfall-example/src/App.js:148]: Use an arrow fn
`formatter: () => {`

File: waterfall-example/src/App.js [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+
+          default:
+            // this number is greater than 100 billion
+            // i'm not going to keep doing this...
+        }
+
+
+        // determine the prefix and the date name to be shown in the chart
+        let prefix = this.y >= 0 ? '$' : '-$';
+        let date_name = this.key;
```
Comment (Pearce-Ropion) [inline waterfall-example/src/App.js:214]: `const`

File: waterfall-example/src/App.js [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+
+        return `${date_name} :    ${prefix + output_num + suffix + ' USD'}`;
```
Comment (Pearce-Ropion) [inline waterfall-example/src/App.js:216]: Why is there so much whitespace here?

File: waterfall-example/src/setupTests.js [hunk 11]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
```
Comment (Pearce-Ropion) [inline waterfall-example/src/setupTests.js:1]: If you are not using tests, remove this

**Pull request closed without merging**